python/triton/compiler.py

```diff
@@ -1546,43 +1561,57 @@ def format_of(ty):
   return ptr_info;
 }}
 
-static PyObject* launch(PyObject* self, PyObject* args) {{
-  int gridX, gridY, gridZ;
-  uint64_t _stream;
-  uint64_t _function;
-  int num_warps;
-  int shared_memory;
-  PyObject *launch_enter_hook = NULL;
-  PyObject *launch_exit_hook = NULL;
-  PyObject *compiled_kernel = NULL;
-  {' '.join([f"{_extracted_type(ty)} _arg{i}; " for i, ty in signature.items()])}
-  if(!PyArg_ParseTuple(args, \"{format}\", &gridX, &gridY, &gridZ, &num_warps, &shared_memory, &_stream, &_function, &launch_enter_hook, &launch_exit_hook, &compiled_kernel, {', '.join(f"&_arg{i}" for i, ty in signature.items())})) {{
-    return NULL;
+static PyObject* launch(PyObject* self, PyObject* const* args, Py_ssize_t nargs, PyObject* kwnames) {{
```

Comment:
`METH_FASTCALL` doesn't have kwargs. See https://docs.python.org/3/c-api/structures.html#c._PyCFunctionFast

```suggestion
static PyObject* launch(PyObject* self, PyObject* const* args, Py_ssize_t nargs) {{
```